Documentation/network/l2-announcements.rst

```diff
@@ -0,0 +1,314 @@
+
+    $ kubectl -n kube-system get lease/cilium-l2announce-default-deathstar -o yaml
+    apiVersion: coordination.k8s.io/v1
+    kind: Lease
+    metadata:
+      creationTimestamp: "2023-05-09T15:13:32Z"
+      name: cilium-l2announce-default-deathstar
+      namespace: kube-system
+      resourceVersion: "449966"
+      uid: e3c9c020-6e24-4c5c-9df9-d0c50f6c4cec
+    spec:
+      acquireTime: "2023-05-09T15:14:20.108431Z"
+      holderIdentity: worker-node
+      leaseDurationSeconds: 3
+      leaseTransitions: 1
+      renewTime: "2023-05-12T12:15:26.773020Z"
+
+The ``acquireTime`` is the time at which the current leader acquired the lease.
+The ``holderIdentity`` is the name of the current holder/leader node. 
+If the leader does not renew the lease for ``leaseDurationSeconds`` seconds a
+new leader is chosen. ``leaseTransitions`` indicates how often the lease changed
+hands and ``renewTime`` the last time the leader renewed the lease.
+
+There are three Helm options that can be tuned with regards to leases:
+
+* ``l2announcements.leaseDuration`` determines the ``leaseDurationSeconds`` value
+  of created leases and by extent how long a leader must be "down" before 
+  failover occurs. (default is 15s)
+
+* ``l2announcements.leaseRenewDeadline`` is the interval at which the leader 
+  should renew the lease. (default is 5s)
+
+* ``l2announcements.leaseRetryPeriod`` if renewing the lease fails, how long 
+  should the agent wait before it tries again. (default is 2s)
+
+.. tabs::
+    .. group-tab:: Helm
+
+        .. parsed-literal::
+
+            $ helm upgrade cilium |CHART_RELEASE| \\
+               --namespace kube-system \\
+               --reuse-values \\
+               --set l2announcements.enabled=true \\
+               --set kubeProxyReplacement=strict \\
+               --set l2announcements.leaseDuration=3s \\
+               --set l2announcements.leaseRenewDeadline=1s \\
+               --set l2announcements.leaseRetryPeriod=200ms
+
+    .. group-tab:: ConfigMap
+
+        .. code-block:: yaml
+
+            enable-l2-announcements: true
+            kube-proxy-replacement: strict
+            l2-announcements-lease-duration: 3s
+            l2-announcements-renew-deadline: 1s
+            l2-announcements-retry-period: 200ms
+
+There is a trade-off between fast failure detection and CPU + network usage. 
+Each service incurs a CPU and network overhead, so clusters with smaller amounts
+of services can more easily afford faster failover times. Larger clusters might
+need to increase parameters if the overhead is to much.
+
+Failover
+--------
+
+When nodes participating in leader election detect that the lease holder did not
+renew the lease for ``leaseDurationSeconds`` amount of seconds, they will ask
+the API server to make them the new holder. The first request to be processed 
+gets through and the rest are denied.
+
+When a node becomes the leader/holder, it will send out a gratuitous ARP reply 
```

Comment:
We should probably mention Unsolicited Neighbor Advertisements as well or at least add context that this is IPv4 only. In the limitation section we already mentioned it currently doesn't support IPv6 so perhaps let's go for the latter?